api4/plugin.go

```diff
@@ -81,6 +83,51 @@ func uploadPlugin(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(manifest.ToJson()))
 }
 
+func installPluginFromUrl(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !*c.App.Config().PluginSettings.Enable {
+		c.Err = model.NewAppError("installPluginFromUrl", "app.plugin.disabled.app_error", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	url := c.Params.PluginDownloadUrl
+
+	if !model.IsValidHttpUrl(url) {
+		c.Err = model.NewAppError("installPluginFromUrl", "api.plugin.install.invalid_url.app_error", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	if !(strings.HasPrefix(url, "https") || *c.App.Config().PluginSettings.AllowInsecureDownloadUrl) {
+		c.Err = model.NewAppError("installPluginFromUrl", "api.plugin.install.insecure_url.app_error", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	resp, err := http.Get(url)
```

Comment:
We should probably use the httpservice to get a client for here so all our regular settings apply. https://github.com/mattermost/mattermost-server/blob/master/app/app.go#L46